src/lib/storage/index/partial_hash/partial_hash_index_impl.cpp

```diff
@@ -126,7 +126,7 @@ size_t PartialHashIndexImpl<DataType>::insert_entries(
 
     // Prevents multiple threads from indexing the same chunk concurrently.
```

Comment:
Is the problem here that multiple writers index the same chunk or that multiple writer insert in parallel? I would expect that the chunk is not relevant as long as there is only a single writer?

Reply:
The problem was that multiple writers index the same chunk concurrently.
There was the possibility that mutliple writers could pass the check if the chunk is already indexed, so a chunk could get indexed multiple times.